src/libmongoc/tests/test-mongoc-client.c

```diff
@@ -571,10 +571,22 @@ test_mongoc_client_speculative_auth_failure (bool pooled)
    if (!r) {
       r = mongoc_cursor_error (cursor, &error);
       BSON_ASSERT (r);
-      ASSERT_ERROR_CONTAINS (error,
-                             MONGOC_ERROR_CLIENT,
-                             MONGOC_ERROR_CLIENT_AUTHENTICATE,
-                             "Failed to send \"saslContinue\" command");
+      /* A client pool on servers supporting speculative auth (4.4+) will get an
+       * error on saslContinue and subsequently attempt to start auth from the
+       * beginning. The failpoint closes the stream causing an error from
+       * saslStart. */
```

Comment:
The pooled connection speculative auth logic will re-use the closed connection to attempt to restart authentication. Single-threaded connections will bail out earlier due to the changes of CDRIVER-3934. 

I filed CDRIVER-4090 to investigate this on pooled connections, since that seems like it could be a bug.

Note, this was not surfacing before because the network error would cause the server to be marked Unknown in the topology description. The attempt to restart auth would error earlier when attempting to assemble a command on an Unknown server [here](https://github.com/kevinAlbs/mongo-c-driver/blob/bcf7da6489bdaf7c0adf295a350b5e0887c67fa1/src/libmongoc/src/mongoc/mongoc-cmd.c#L835:L839). That appended the previous error message containing `Failed to send \"saslContinue\"`.